lib/puppet/type/iis_site.rb

```diff
@@ -0,0 +1,270 @@
+require 'puppet/parameter/boolean'
+
+Puppet::Type.newtype(:iis_site) do
+  @doc = "Create a new IIS website."
+
+  newproperty(:ensure) do
+    desc "Specifies whether a site should be present or absent. If present is
+      specified, the site will be created but left in the default stopped state.
+      If started is specified, then the site will be created as well as started.
+      If stopped is specified, then the site will be created and kept stopped."
+
+    newvalue(:stopped) do
+      provider.stop
+    end
+
+    newvalue(:started) do
+      provider.start
+    end
+
+    newvalue(:present) do
+      provider.create
+    end
+
+    newvalue(:absent) do
+      provider.destroy
+    end
+
+    aliasvalue(:false, :stopped)
+    aliasvalue(:true, :started)
+  end
+
+  newparam(:name, :namevar => true) do
+    desc "The Name of the IIS site. Used for uniqueness. Will set
+      the target to this value if target is unset."
+
+    validate do |value|
+      if value.nil? or value.empty?
+        raise ArgumentError, "A non-empty name must be specified."
+      end
+      fail("#{name} is not a valid web site name") unless value =~ /^[a-zA-Z0-9\-\_'\s]+$/
+    end
+  end
+
+  newproperty(:physicalpath) do
+    desc 'The physical path to the IIS web site folder'
+    validate do |value|
+      if value.nil? or value.empty?
+        raise ArgumentError, "A non-empty physicalpath must be specified."
+      end
+      fail("File paths must be fully qualified, not '#{value}'") unless value =~ /^.:(\/|\\)/ or value =~ /^\/\/[^\/]+\/[^\/]+/
```

Comment:
If I'm understanding correctly, in `/^.:(\/|\\)/` is it worth restricting the first character in case the user accidentally typos this (ex: `physicalpath => '@:/mysite'`)? May not be worth it if we don't get far.